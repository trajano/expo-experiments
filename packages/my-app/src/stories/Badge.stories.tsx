import type { Meta, StoryObj } from '@storybook/react';
import * as Notifications from 'expo-notifications';
import { FC, useCallback, useEffect, useState } from 'react';
import { Button, StyleSheet, View } from 'react-native';
import { MyText, MyTextInput, Strong } from 'react-native-my-text';

const NotificationBadgesView: FC = () => {
  const [currentBadgeCount, setCurrentBadgeCount] = useState(0);
  const [nextBadgeCountData, setNextBadgeCountData] = useState(0);
  const [badgeCountUpdated, setBadgeCountUpdated] = useState(true);
  const onRefreshBadgeCount = useCallback(() => {
    (async () => {
      const nextBadgeCount = await Notifications.getBadgeCountAsync();
      setCurrentBadgeCount(nextBadgeCount);
    })();
  }, []);
  const onIncrementBadgeCount = useCallback(() => {
    (async () => {
      const nextBadgeCount = (await Notifications.getBadgeCountAsync()) + 1;
      const updated = await Notifications.setBadgeCountAsync(nextBadgeCount);
      setCurrentBadgeCount(nextBadgeCount);
      setNextBadgeCountData(nextBadgeCount);
      setBadgeCountUpdated(updated);
    })();
  }, []);
  const onClearBadgeCount = useCallback(() => {
    (async () => {
      const updated = await Notifications.setBadgeCountAsync(0);
      setCurrentBadgeCount(0);
      setNextBadgeCountData(0);
      setBadgeCountUpdated(updated);
    })();
  }, []);
  useEffect(() => {
    let mounted = true;
    (async () => {
      const nextBadgeCount = await Notifications.getBadgeCountAsync();
      if (mounted) {
        setCurrentBadgeCount(nextBadgeCount);
        setNextBadgeCountData(nextBadgeCount);
      }
    })();
    return () => {
      mounted = false;
    };
  }, []);

  const onSetBadgeCount = useCallback(() => {
    (async () => {
      const nextUpdated =
        await Notifications.setBadgeCountAsync(nextBadgeCountData);
      setBadgeCountUpdated(nextUpdated);
    })();
  }, [nextBadgeCountData]);

  return (
    <View style={styles.container}>
      <MyText>
        <Strong>Current badge count:</Strong>
        <MyText testID="current-badge-count">{currentBadgeCount}</MyText>
      </MyText>
      <MyText>
        <Strong>Badge count updated:</Strong> {badgeCountUpdated ? 'yes' : 'no'}
      </MyText>
      <MyText>
        <Strong>Next badge count:</Strong>
      </MyText>
      <MyTextInput
        style={{
          fontSize: 48,
          borderColor: 'black',
          borderWidth: 1,
          paddingHorizontal: 16,
        }}
        keyboardType="numeric"
        textAlign="right"
        onChangeText={(nextText) => {
          try {
            const nextBadgeCount = parseInt(nextText);
            if (isNaN(nextBadgeCount) || nextBadgeCount < 0) {
              setNextBadgeCountData(0);
            } else {
              setNextBadgeCountData(nextBadgeCount);
            }
          } catch (e: unknown) {
            setNextBadgeCountData(0);
          } finally {
            setBadgeCountUpdated(false);
          }
        }}
        defaultValue={nextBadgeCountData.toString(10)}
      />
      <Button
        disabled={badgeCountUpdated}
        title="set badge count"
        onPress={onSetBadgeCount}
      />
      <Button
        title="refresh badge count"
        testID="refresh-badge-count-button"
        onPress={onRefreshBadgeCount}
      />
      <Button title="increment badge count" onPress={onIncrementBadgeCount} />
      <Button title="clear badge count" onPress={onClearBadgeCount} />
    </View>
  );
};

const styles = StyleSheet.create({
  outerBox: {
    borderWidth: 1,
    borderColor: 'red',
  },
  container: {
    padding: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  text: {
    fontSize: 16,
    fontFamily: 'Nunito',
  },
});

const meta: Meta<typeof NotificationBadgesView> = {
  title: 'Badges',
  component: NotificationBadgesView,
  parameters: {
    notes: 'This set badges.',
  },
};

export default meta;

type Story = StoryObj<typeof NotificationBadgesView>;

export const Default: Story = {
  args: {},
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};
