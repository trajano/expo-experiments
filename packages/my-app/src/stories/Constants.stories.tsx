import type { Meta, StoryObj } from '@storybook/react';
import * as ExpoApplication from 'expo-application';
import ExpoConstants from 'expo-constants';
import * as ExpoFileSystem from 'expo-file-system';
import * as Localization from 'expo-localization';
import * as ExpoManifests from 'expo-manifests';
import * as ExpoUpdates from 'expo-updates';
import { FC } from 'react';
import { Platform as RNPlatform, StyleSheet, View } from 'react-native';
import { MyText } from 'react-native-my-text';
const ConstantsView: FC<{ content: object }> = ({ content }) => {
  return (
    <View style={styles.container}>
      <MyText style={styles.text}>{JSON.stringify(content, null, 2)}</MyText>
    </View>
  );
};

const styles = StyleSheet.create({
  outerBox: {
    borderWidth: 1,
    borderColor: 'red',
  },
  container: {
    padding: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  text: {
    fontSize: 16,
    fontFamily: 'Nunito',
  },
});

const meta: Meta<typeof ConstantsView> = {
  title: 'Constants',
  component: ConstantsView,
  parameters: {
    notes: 'This checks constant values',
  },
};

export default meta;

type Story = StoryObj<typeof ConstantsView>;

export const Constants: Story = {
  args: { content: ExpoConstants },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const ConstantsSubset: Story = {
  args: {
    content: {
      appOwnership: ExpoConstants.appOwnership,
      expoConfig: ExpoConstants.expoConfig,
    },
  },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const Application: Story = {
  args: { content: ExpoApplication },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const Updates: Story = {
  args: { content: ExpoUpdates },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const Calendars: Story = {
  args: { content: Localization.getCalendars() },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const Locales: Story = {
  args: { content: Localization.getLocales() },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const FileSystem: Story = {
  args: { content: ExpoFileSystem },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const Manifests: Story = {
  args: { content: ExpoManifests },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};

export const Platform: Story = {
  args: { content: RNPlatform },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};
