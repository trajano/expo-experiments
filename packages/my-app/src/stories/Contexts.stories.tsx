import { useMyBackgroundFetch } from '@/hooks/MyBackgroundFetch';
import { useUserPreferences } from '@/hooks/UserPreferences';
import type { Meta, StoryObj } from '@storybook/react';
import { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { useNotifications } from 'react-native-my-hooks';
import { MyText } from 'react-native-my-text';
const ContextsView: FC = () => {
  const myBackgroundFetch = useMyBackgroundFetch();
  const notifications = useNotifications();
  const userPreferences = useUserPreferences();
  return (
    <View style={styles.container}>
      <View style={styles.sectionHeader}>
        <MyText style={styles.sectionHeaderText}>useMyBackgroundFetch</MyText>
      </View>
      <MyText style={styles.text}>
        {JSON.stringify(myBackgroundFetch, null, 2)}
      </MyText>
      <View style={styles.sectionHeader}>
        <MyText style={styles.sectionHeaderText}>useNotifications</MyText>
      </View>
      <MyText style={styles.text}>
        {JSON.stringify(notifications, null, 2)}
      </MyText>
      <View style={styles.sectionHeader}>
        <MyText style={styles.sectionHeaderText}>useUserPreferences</MyText>
      </View>
      <MyText style={styles.text}>
        {JSON.stringify(userPreferences, null, 2)}
      </MyText>
    </View>
  );
};

const styles = StyleSheet.create({
  outerBox: {
    borderWidth: 1,
    borderColor: 'red',
  },
  container: {
    padding: 20,
  },
  sectionHeader: {
    padding: 10,
    backgroundColor: 'black',
  },
  sectionHeaderText: {
    fontWeight: 'bold',
    color: 'white',
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  text: {
    fontSize: 16,
    fontFamily: 'Nunito',
  },
});
const meta: Meta<typeof ContextsView> = {
  title: 'Contexts',
  component: ContextsView,
  parameters: {
    notes: 'This checks the value of custom contexts that have been used.',
  },
};

export default meta;

type Story = StoryObj<typeof ContextsView>;

export const Default: Story = {
  args: {},
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};
