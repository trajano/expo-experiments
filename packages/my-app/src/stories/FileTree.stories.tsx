import { PreviewViewMode } from '@sb/preview';
import type { Meta, StoryObj } from '@storybook/react';
import * as FileSystem from 'expo-file-system';
import { FileTree } from 'react-native-my-components';

const meta: Meta<typeof FileTree> = {
  title: 'FileTree',
  component: FileTree,
};

export default meta;

type Story = StoryObj<typeof FileTree>;

export const DocumentDirectory: Story = {
  args: { directoryUri: FileSystem.documentDirectory! },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
    previewViewMode: PreviewViewMode.NO_SCROLL_VIEW,
  },
};

export const CacheDirectory: Story = {
  args: { directoryUri: FileSystem.cacheDirectory! },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
    previewViewMode: PreviewViewMode.NO_SCROLL_VIEW,
  },
};

export const BundleDirectory: Story = {
  args: { directoryUri: FileSystem.bundleDirectory! },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
    previewViewMode: PreviewViewMode.NO_SCROLL_VIEW,
  },
};
