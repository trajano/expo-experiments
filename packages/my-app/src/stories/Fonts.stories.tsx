import _ from 'lodash';
import { FlashList, ListRenderItem } from '@shopify/flash-list';
import type { Meta, StoryObj } from '@storybook/react';
import ExpoConstants from 'expo-constants';
import { PreviewViewMode } from '@sb/preview';
import { FC, useCallback, useMemo } from 'react';
import { StyleSheet, Text, TextStyle, View } from 'react-native';
import { MyText } from 'react-native-my-text';

const FontRowView: FC<{
  fontFamilyName: string;
  fontSize: TextStyle['fontSize'];
  fontWeight: TextStyle['fontWeight'];
  testString: string;
}> = ({ fontFamilyName, fontSize, fontWeight, testString }) => {
  return (
    <>
      <View style={styles.sectionHeader}>
        <MyText style={styles.sectionHeaderText}>
          {fontFamilyName} {fontWeight}
        </MyText>
      </View>
      <Text
        style={{
          fontFamily: fontFamilyName,
          fontSize: fontSize,
          fontWeight: fontWeight,
        }}
      >
        {testString}
      </Text>
    </>
  );
};
const Separator: FC = () => <View style={styles.separator} />;

/**
 * Fonts I really want to see.
 */
const importantFonts = [
  'Arial',
  'Roboto',
  'Nunito',
  'Noto',
  'SF Pro',
  'sans-serif',
];

const FontView: FC<{
  fontSize: TextStyle['fontSize'];
  testString: string;
  fontWeight: TextStyle['fontWeight'];
  importantFontsOnly: boolean;
}> = ({ fontSize, fontWeight, testString, importantFontsOnly }) => {
  const renderItem = useCallback<ListRenderItem<string>>(
    ({ item }) => (
      <FontRowView
        fontFamilyName={item}
        fontSize={fontSize}
        fontWeight={fontWeight}
        testString={testString}
      />
    ),
    [fontSize, fontWeight, testString],
  );
  const fontsToRender = useMemo<string[]>(() => {
    const fontsToPrepend = importantFonts.filter((font) =>
      ExpoConstants.systemFonts.includes(font),
    );
    if (importantFontsOnly) {
      return fontsToPrepend;
    }
    return _.uniq([...fontsToPrepend, ...ExpoConstants.systemFonts]);
  }, [importantFontsOnly]);

  return (
    <FlashList
      contentContainerStyle={styles.container}
      data={fontsToRender}
      estimatedItemSize={100}
      ItemSeparatorComponent={Separator}
      renderItem={renderItem}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
  sectionHeader: {
    padding: 10,
    backgroundColor: 'black',
  },
  sectionHeaderText: {
    fontWeight: 'bold',
    color: 'white',
  },
  separator: {
    height: 10,
  },
});

const fontWeights = [
  'normal',
  'bold',
  '100',
  '200',
  '300',
  '400',
  '500',
  '600',
  '700',
  '800',
  '900',
];
const meta: Meta<typeof FontView> = {
  title: 'Font',
  component: FontView,
  argTypes: {
    fontSize: {},
    fontWeight: {
      options: fontWeights,
      control: {
        type: 'select',
      },
    },
  },
  parameters: {
    notes: 'This shows the native system fonts.',
  },
};

export default meta;

type Story = StoryObj<typeof FontView>;

export const QuickBrownFox: Story = {
  args: {
    fontSize: 16,
    fontWeight: '400',
    importantFontsOnly: false,
    testString: 'The quick brown fox jumped over the lazy dogs.',
  },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
    previewViewMode: PreviewViewMode.NO_SCROLL_VIEW,
  },
};

export const JordanPeterson: Story = {
  args: {
    fontSize: 16,
    fontWeight: '400',
    importantFontsOnly: true,
    // This is a ChatGPT response to
    // give me a string in the style of a Jordan Peterson quote that would use as much of the alphabet as possible, numbers would also be a bonus.
    testString:
      'To seek clarity in a world that thrives on chaos, you must position yourself as a beacon of order, illuminating the path from A to Z, even if that means deciphering 1,000 paradoxes along the way, embracing both the known and the unknown, the x-factor and the y, till every question mark morphs into the solid ground of understanding.',
  },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
    previewViewMode: PreviewViewMode.NO_SCROLL_VIEW,
  },
};

export const AlexJones: Story = {
  args: {
    fontSize: 16,
    fontWeight: '700',
    importantFontsOnly: true,
    // This is a ChatGPT response to
    // give me a string in the style of a Jordan Peterson quote that would use as much of the alphabet as possible, numbers would also be a bonus.
    testString:
      'They‚Äôre watching you 24/7! üïµÔ∏è‚Äç‚ôÇÔ∏èüëÅÔ∏è From A to Z, they‚Äôre using every tool‚Äîüíªüì≤üîç‚Äîto control your life! Wake up before it‚Äôs too late! üö®üî• 1% of people see the truth, but the rest? Asleep! üí§üí°',
  },
  parameters: {
    backgrounds: {
      default: 'plain',
    },
    previewViewMode: PreviewViewMode.NO_SCROLL_VIEW,
  },
};
