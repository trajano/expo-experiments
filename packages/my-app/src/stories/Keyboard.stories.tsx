import type { Meta, StoryObj } from '@storybook/react';

import { FC, useEffect, useState } from 'react';
import { Keyboard, StyleSheet } from 'react-native';
import { MyText as Text, MyTextInput as TextInput } from 'react-native-my-text';

const KeyboardSample: FC = () => {
  const [keyboardStatus, setKeyboardStatus] = useState('Keyboard Hidden');

  useEffect(() => {
    const showSubscription = Keyboard.addListener('keyboardDidShow', () => {
      setKeyboardStatus('Keyboard Shown');
    });
    const hideSubscription = Keyboard.addListener('keyboardDidHide', () => {
      setKeyboardStatus('Keyboard Hidden');
    });

    return () => {
      showSubscription.remove();
      hideSubscription.remove();
    };
  }, []);

  return (
    <>
      <TextInput
        style={style.input}
        placeholder="Click hereâ€¦"
        onSubmitEditing={Keyboard.dismiss}
      />
      <Text style={style.status}>{keyboardStatus}</Text>
    </>
  );
};

const meta: Meta<typeof KeyboardSample> = {
  title: 'Keyboard',
  component: KeyboardSample,
  parameters: {
    notes: 'Nothing.',
  },
};

export default meta;

type Story = StoryObj<typeof KeyboardSample>;

export const Default: Story = {
  args: {},
  parameters: {},
};
const style = StyleSheet.create({
  container: {
    flex: 1,
    padding: 36,
  },
  input: {
    padding: 10,
    borderWidth: 0.5,
    borderRadius: 4,
  },
  status: {
    padding: 16,
    textAlign: 'center',
  },
});
