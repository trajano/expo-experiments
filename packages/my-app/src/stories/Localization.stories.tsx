import type { Meta, StoryObj } from '@storybook/react';
import * as Localization from 'expo-localization';
import { FC, useMemo } from 'react';
import { StyleSheet, View } from 'react-native';
import JSONTree, { JSONTreeProps } from 'react-native-json-tree';
import { MyText, Strong } from 'react-native-my-text';

const labelRenderer: JSONTreeProps['labelRenderer'] = (keypath) => (
  <MyText style={{ fontSize: 20 }}>{keypath[0]}</MyText>
);
const valueRenderer: JSONTreeProps['valueRenderer'] = (raw) => (
  <MyText style={{ fontSize: 16 }}>
    {typeof raw === 'string' ? raw : JSON.stringify(raw)}
  </MyText>
);
const ExpoLocalizationView: FC = () => {
  const locales = Localization.useLocales();
  const calendars = Localization.useCalendars();
  const localesFromGet = useMemo(() => Localization.getLocales(), []);
  const calendarsFromGet = useMemo(() => Localization.getCalendars(), []);
  return (
    <View style={styles.container}>
      <MyText>
        <Strong>Timezone:</Strong>
        {calendars[0].timeZone}
      </MyText>
      <MyText>
        <Strong>useCalendars</Strong>
      </MyText>
      <JSONTree
        data={calendars}
        shouldExpandNode={() => true}
        hideRoot={true}
        labelRenderer={labelRenderer}
        valueRenderer={valueRenderer}
      />
      <MyText>
        <Strong>useLocales</Strong>
      </MyText>
      <JSONTree
        data={locales}
        shouldExpandNode={() => true}
        hideRoot={true}
        labelRenderer={labelRenderer}
        valueRenderer={valueRenderer}
      />
      <MyText>
        <Strong>getCalendars</Strong>
      </MyText>
      <JSONTree
        data={calendarsFromGet}
        shouldExpandNode={() => true}
        hideRoot={true}
        labelRenderer={labelRenderer}
        valueRenderer={valueRenderer}
      />
      <MyText>
        <Strong>getLocales</Strong>
      </MyText>
      <JSONTree
        data={localesFromGet}
        shouldExpandNode={() => true}
        hideRoot={true}
        labelRenderer={labelRenderer}
        valueRenderer={valueRenderer}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  outerBox: {
    borderWidth: 1,
    borderColor: 'red',
  },
  container: {
    padding: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  text: {
    fontSize: 16,
    fontFamily: 'Nunito',
  },
});

const meta: Meta<typeof ExpoLocalizationView> = {
  title: 'Expo Localization',
  component: ExpoLocalizationView,
  parameters: {
    notes: 'This checks expo-localization information.',
  },
};

export default meta;

type Story = StoryObj<typeof ExpoLocalizationView>;

export const Default: Story = {
  args: {},
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};
