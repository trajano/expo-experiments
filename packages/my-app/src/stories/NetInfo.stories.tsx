import { useNetInfo, addEventListener } from '@react-native-community/netinfo';
import type { Meta, StoryObj } from '@storybook/react';
import { FC, useEffect, useState } from 'react';
import { StyleSheet, View } from 'react-native';
import { MyText, Strong } from 'react-native-my-text';
import * as Network from 'expo-network';

const NetInfoView: FC = () => {
  const netInfoState = useNetInfo();
  const [ipAddress, setIpAddress] = useState('');
  const [networkState, setNetworkState] = useState<Network.NetworkState>({
    type: Network.NetworkStateType.UNKNOWN,
  });
  useEffect(() => {
    let mounted = true;
    const unsubscribe = addEventListener(async () => {
      // though this can also update netInfoState, useNetInfo is utilized.
      // this effect is primarily to get the values from expo-network that
      // does not have a listener pattern.
      const nextIpAddress = await Network.getIpAddressAsync();
      const nextNetworkState = await Network.getNetworkStateAsync();
      if (mounted) {
        setIpAddress(nextIpAddress);
        setNetworkState(nextNetworkState);
      }
    });
    return () => {
      mounted = false;
      unsubscribe();
    };
  }, []);
  return (
    <View style={styles.container}>
      <MyText style={styles.text}>
        {JSON.stringify(netInfoState, null, 2)}
      </MyText>
      <MyText style={styles.text}>
        <Strong>IP Address</Strong>: {ipAddress}
      </MyText>
      <MyText style={styles.text}>
        <Strong>Network State:</Strong>{' '}
        {Network.NetworkStateType[networkState.type!]}
      </MyText>
      <MyText style={styles.text}>
        {JSON.stringify(networkState, null, 2)}
      </MyText>
    </View>
  );
};

const styles = StyleSheet.create({
  outerBox: {
    borderWidth: 1,
    borderColor: 'red',
  },
  container: {
    padding: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  text: {
    fontSize: 16,
    fontFamily: 'Nunito',
  },
});

const meta: Meta<typeof NetInfoView> = {
  title: 'NetInfo',
  component: NetInfoView,
  parameters: {
    notes: 'This checks Network information.',
  },
};

export default meta;

type Story = StoryObj<typeof NetInfoView>;

export const Default: Story = {
  args: {},
  parameters: {
    backgrounds: {
      default: 'plain',
    },
  },
};
