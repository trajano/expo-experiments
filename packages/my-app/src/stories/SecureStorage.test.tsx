import { composeStories } from '@storybook/react';
import { act, render, screen } from '@testing-library/react-native';
import * as stories from './SecureStorage.stories';

const { Default } = composeStories(stories);
test('only show minimal fonts', async () => {
  render(<Default />);
  await act(() => Promise.resolve());
  expect(screen.toJSON()).toMatchSnapshot();
});
