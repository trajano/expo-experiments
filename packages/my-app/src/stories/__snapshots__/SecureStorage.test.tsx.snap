// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`only show minimal fonts 1`] = `
<View>
  <View
    style={
      {
        "backgroundColor": "black",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": undefined,
          "fontStyle": undefined,
          "fontWeight": "bold",
          "lineHeight": undefined,
        }
      }
    >
      secure
    </Text>
  </View>
  <Text
    style={
      {
        "fontFamily": undefined,
        "fontStyle": undefined,
        "fontWeight": undefined,
        "lineHeight": undefined,
      }
    }
  />
  <Text
    style={
      {
        "fontFamily": undefined,
        "fontStyle": undefined,
        "fontWeight": undefined,
        "lineHeight": undefined,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": undefined,
          "fontStyle": undefined,
          "fontWeight": "bold",
          "lineHeight": undefined,
        }
      }
    />
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        refresh state
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        clear value
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        increment value
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
          {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        clear error
      </Text>
    </View>
  </View>
</View>
`;
